draft-ietf-tls-ctls.md

```diff
@@ -450,15 +466,23 @@ framing, except for two changes:
               case finished:              Finished;
               case new_session_ticket:    NewSessionTicket;
               case key_update:            KeyUpdate;
+              case request_connection_id: RequestConnectionId;
+              case new_connection_id:     NewConnectionId;
           };
-      } Handshake;
+      } CTLSHandshake;
 ~~~~
 
+Each `CTLSHandshake` MUST be conveyed as a single `CTLSPlaintext.fragment`
+or `CTLSCiphertext.encrypted_record`, and is therefore limited to a maximum
+length of `2^16-1`.  When operating over UDP, large `CTLSHandshake` messages
+will also require the use of IP fragmentation, which is sometimes
+undesirable.  Operators can avoid these concerns by setting
+`template.handshakeFraming = true`.
```

Comment:
Do we need to say how you also need an in-order transport?

Reply:
I had intended this to be usable even on out-of-order transports.  It appears to be safe for DTLS 1.3, because the only messages sent in plaintext are ClientHello, ServerHello, HelloRetryRequest, and Alert, each of which constitutes an entire "flight" and so is at no risk of reordering.  All multi-message flights are in ciphertext, and can rely on the unified header's sequence number for ordering.

This would not have worked in DTLS 1.2, and I suppose it is not guaranteed for future versions.  Perhaps we can nonetheless rely on this property for now, and address the problem if it arises in the future.

I'm not 100% sure how this interacts with the DTLS ACK and retransmission system, but it seems reasonably straightforward to deduplicate identical ServerHellos, to avoid placing a retransmitted ServerHello in the transcript twice.

Reply:
I don't think you can rely on the sequence number for reordering because retransmits use a new sequence number.

Reply:
I had been thinking that we would change that, but never mind.

I've added a `CTLSDatagramHandshake` type that has a `message_seq` field for reordering.